Source/Shaders/ModelExperimental/ModelExperimentalVS.glsl

```diff
@@ -40,6 +33,21 @@ void main()
     customShaderStage(vsOutput, attributes);
     #endif
 
+    // Update the position for this instance in place
+    #ifdef HAS_INSTANCING
```

Comment:
Should this block be put back above `customShaderStage` so that users can modify each instance?

Reply:
Based on offline discussion, I moved the instancing shader code before the custom shader. Here are some examples of custom shaders in action, operating on each feature using `a_instanceFeatureId_<N>`:

|[`glTF`](http://localhost:8080/Apps/Sandcastle/index.html#c=jVRtT9swEP4rVj6lW3FSKC8rBU2UlyGBQCtiH5YJmeTSeDh2ZDulgPjvO9tpef0wCZXz+XnOd/fcZc40mXO4B032iIR7MgHD25pee1+cRbk/T5S0jEvQWdTbzWQm58hrlOGWK4nMjjVh2qLF5AYttaoPYaYBTJxJQtYG6xs03R4Otwbf+s4xHNJ0M93YTrf8cZCmmXSxXeSqeZfOD2AFl7NLbvPqpxIiTvvE/b0kU/IFFMea1XClmTSl0vVLXiuXoULlTAScOl5xTgBrY1Zpn2sWSaVtlUX9cLoHY7MopBdeg0XDZHGuChDTihXvuzdpjVV1uImfXJA5YGcWwXMFCzsiv52bkCQhpyWxFZBatQYIN6TQbDaDgljl/ZrPKtvvIPggmWl1b/6XLaB8QzaV5vKuo2fRXPGiS+4cBY6vvXkqm9aSufH/+4RLhcf8sb7xMQLmorUBFIxe1y0X9OmV7X6XGLqcmPMJ+bKHz+YbMbvh0lgmczgGZlsNp8VNSr6SAXXivsR57uw/9K/CPFEO1CLqoe/5RZZQ4l430dTkKCttNK/x1TkYyorCC9zJ5PU7WjSACMABF35qT4Qtg2iEzNAe4evJtIHcJIfMssSzTOJgZ8rpmRwo7Fgookhm4uo4uVWLtWVdBXVRVqX4HM+Z1Xwx+mQ+q3eT/npK4xCBrDavv3TgwqzsLuaRELwxKC/9dTLdGa6uP1mUcNVbJsglr9v6EnFiyh9hRAbrO91d/mqwRx+3wIGee8tFyaSvlWos6eESO8sNUJxEGZetzP2XI/aIHgk7ElTLMTXNaCkertSBaqVrx7SpQENA09s3zj7ptCpa3GAMOiIp3QyZ7PrpiPrR2NgHAfvLHnzndYMbTlotYkoTC3UjGH65kts2vwNLc2Mc10HHyWvquOBzwou9T76LJBfMGLwpW+H7lkX74wTxH6hCeYkvcO0Ee3CwarB/FpyU0nGCx8+ZVilxy/S7yP8A) | [`I3DM`](http://localhost:8080/Apps/Sandcastle/index.html#c=rVVZb+M2EP4rhF8iLxTK8SZY1PEGRbwpmqJBFrWRPqyDgJbGFjcUKfCwc8D/vUPqjDc9HhoEFjmc8/uGwxkY7gp69VSC5gVIy8QvwKzTYChIthJwozIQ/XPymVjt4Hwpl3LLNNly2IFGqYQdmVX+7oIsWg7SsJ8pNOQS9HIwrOzCOU2FSh9p6rRGzwv0/9bLb05wJr8wC9H49OzT2XhMT85OP/00HrbBTQrSW9UOw/a8OrJcgAHbqsJTyWQWypnnLDtMeeaMVUV1Er0uJSFb0BaeKskCnuyEfPNiQpKEXK+JzYEUyhkg3JBMs80GMmJVkGu+yW1cq2BAstFqZ/6rtYD1G2OTay4fa/PlYKt4Vid3g6hGd2F5LUtnydaEb0y4VLhNX4qH4KPSuXW2UqoWw+Ugbpy+9tb+t9GhpTLcciVvZuQD4gzpx2hER6fkA2EPXBrLZAp1y1xnDyOkpnO0r9f39LvCRJeDJZIxGKJsX1HYdVDgBRkJyNfUmYB45ey1+hAiWQET9H1dx87+5Da/ZDbNF75d2+iEYA8rp1OYNAKfEqUJ/s9ZUQrAzmJJRf/HLwsfM2m9Jn/jP6lzo9+Nkm2wffxvWd5q7m+PR/L/T7Hn/J/yuw+/BgSkFrJFpTghTmawxtuZeV4CLfWVeJR4PbGNqNUsfYxamrrrZ5USK+ZvUqZS5+cD3YC9EuCXl8/XGXJe61RX/9AzK0vxfMllxuXGdBHixvPwvXx8FT7O7cqA3npW+pbLwUGFGDlYGLcyqeYriNZOph4sEqnSf82wYY6vSVRHq0GJakCHw47dMGZoiRMRr8YWR6WGQm2hVT2vUW+at5YfzKm+A5YhVE3v9IZSn3mw0WvXJ06LCanTp00fxd15wZ544Yp5qgHkvGQpXGmt9ISMekoZrNxmnqvdpXKBgzslnO9cP+F7emlvOE5+nKSN4n5YrSrWepVjgix7/qpVwQ002hQnnOxx0QOYNEC9KFUsVNTJW5dxX9aD7FeMhIV85Xhj/2ByA2+MyZvy/d/xmB6KbpjNKeIXnYxw1JHjtopVjdK8zEED1RjJmZig0rDvobfpkOjQwcoVlq53iEWvfPD0vMHA5vhqkCA/75yE5b57RwMP9KDn61arzppx+m10H6wG8WBq7LOAi8brz7wolba+qSKcPRZw9uCza5KVSx+x8NSYpqmnSd90mvEt4dnnd156kgpmDJ6snRBz/oKj+WKaoP4PpkIFym7xTRPs2avlJxe/V0JK6TTB7fuW7WxpK5lWQJAMB+fxCgcLatW3ZNK+K3Fzcapn/chP66OYbJlw2PsHULbKMxa+qD/LlcKXm5FaBZM8CtVVpk2avZT/Ag)|
| - | - |
| ![image](https://user-images.githubusercontent.com/5172619/145606936-834448ef-f9c6-454f-a731-277c35b79415.png) | ![image](https://user-images.githubusercontent.com/5172619/145607297-dd4a75dc-af4f-4e2c-8d6a-ada57d7ba1b8.png) |

Note that the clipping that occurs in the glTF Sandcastle is due to the model scaling in the shader without scaling its bounding volume.

Reply:
There's no API for it, just yet. I'm assuming the Instance Feature ID will be added to the custom shader input struct at some point in the future.